transmart-rest-api/grails-app/services/org/transmartproject/rest/dataExport/ExportService.groovy

```diff
@@ -0,0 +1,170 @@
+    }
+
+    def exportData(List<Long> ids, String typeOfSet, List types, User user, Long jobId) {
+        def job = jobById(jobId)
+        if (job.jobStatus != JobStatus.CREATED.value) {
+            throw new InvalidRequestException("Job with id $jobId has invalid status. " +
+                    "Expected: $JobStatus.CREATED.value, actual: $job.jobStatus")
+        }
+
+        def dataMap = createJobDataMap(ids, typeOfSet, types, user, jobId, job.jobName)
+        executeExportJob(dataMap)
+    }
+
+    def downloadFile(Long jobId) {
+        def job = jobById(jobId)
+        if(job.jobStatus != JobStatus.COMPLETED.value) {
+            throw new InvalidRequestException("Job with a name is not completed. Current status: '$job.jobStatus'")
+        }
+
+        def exportJobExecutor = new ExportJobExecutor()
+        return exportJobExecutor.getExportJobFileStream(job.viewerURL)
+    }
+
+    def listJobs(User user) {
+        exportAsyncJobService.getJobList(user)
+    }
+
+    Boolean isJobNameUniqueForUser(String name, User user) {
+        jobByNameAndUser(name, user) == null
+    }
+
+    def jobByNameAndUser(String jobName, User user) {
+        exportAsyncJobService.getJobByNameAndUser(jobName, user)
+    }
+
+    def jobById(Long id) {
+        exportAsyncJobService.getJobById(id)
+    }
+
+    def jobUser(Long id) {
+        exportAsyncJobService.getJobUser(id)
+    }
+
+    def isUserAllowedToExport(List<Long> resultSetIds, User user, String typeOfSet) {
+        if (typeOfSet == SupportedTypesOfSet.PATIENT.value) {
+            restDataExportService.patientSetsExportPermission(resultSetIds, user)
+        }
+    }
+
+    List getDataFormats(String typeOfSet, List<Long> setIds, User user) {
+
+        // clinical data (always included)
+        List<String> dataFormats = []
+        dataFormats.add(clinicalDataType)
+
+        // highDim data
+        switch (typeOfSet) {
+            case SupportedTypesOfSet.PATIENT.value:
+                List highDimDataTypes = restDataExportService.highDimDataTypesForPatientSets(setIds, user)
+                if (highDimDataTypes) dataFormats.addAll(highDimDataTypes)
+                break
+            case SupportedTypesOfSet.OBSERVATION.value:
+                List highDimDataTypes = restDataExportService.highDimDataTypesForObservationSets(setIds, user)
+                if (highDimDataTypes) dataFormats.addAll(highDimDataTypes)
+                break
+            default:
+                throw new NotSupportedException("Set type '$typeOfSet' not supported.")
+        }
+
+        dataFormats
+    }
+
+    private AsyncJob executeExportJob(Map dataMap) {
+
+        def job
+        def jobDataMap = new JobDataMap(dataMap)
+        JobDetail jobDetail = JobBuilder.newJob(ExportJobExecutor.class)
+                .withIdentity(dataMap.jobId.toString(), 'DataExport')
+                .setJobData(jobDataMap)
+                .build()
+        def randomDelay = Math.random()*10 as int
```

Comment:
I am not sure, this was added by Gijs in [f84542c](https://github.com/thehyve/transmart-core/commit/f84542cc19aa3db6ef02dbe026481ed2368ccfa1) commit to fix job schedulers in transmartApp